File: src/main/java/com/dersaun/apigestaocontas/api/controllers/FaturaController.java
package com.dersaun.apigestaocontas.api.controllers;

import com.dersaun.apigestaocontas.api.request.BuscarLancamentosFaturaRequestDTO;
import com.dersaun.apigestaocontas.domain.FaturaItem;
import com.dersaun.apigestaocontas.domain.dtos.PaginationDTO;
import com.dersaun.apigestaocontas.domain.services.FaturaService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

@RestController
@RequestMapping("faturas")
public class FaturaController {

    @Autowired
    private FaturaService faturaService;

    @GetMapping("fatura-atual")
    public ResponseEntity<Page<FaturaItem>> buscarItensFaturaAtual(@Valid PaginationDTO paginationDTO) {
        var itens = faturaService.buscarItensFaturaAtual(paginationDTO);
        return ResponseEntity.ok(itens);
    }

    @GetMapping("buscar-itens-fatura")
    public ResponseEntity<Page<FaturaItem>> buscarItensFatura(@Valid BuscarLancamentosFaturaRequestDTO paginationDTO) {
        var itens = faturaService.buscarItensFatura(paginationDTO);
        return ResponseEntity.ok(itens);
    }
}

File: src/app/fatura/controllers/fatura.controller.ts
import { Controller, Get, Inject, Post, Query } from '@nestjs/common';
import { Page } from '@utils/dtos/page.dto';
import { FaturaItem } from '@app/fatura/models/fatura-item.entity';
import { AppPaginationDTO } from '@utils/dtos/pagination.dto';
import { IFaturaService } from '@app/fatura/interfaces/fatura.service.interface';
import { BuscarLancamentosFaturaRequestDTO } from '@app/fatura/dtos/buscar-lancamentos-fatura-request.dto';
import { ConsultaItensFaturaResponseDTO } from '@app/fatura/dtos/consulta-itens-fatura-response.dto';
import { User } from '@decorators/usuario.decorator';
import { UsuarioAutenticadoDto } from '@app/autenticacao/dtos/usuario-autenticado.dto';

@Controller({ version: '1', path: 'faturas' })
export class FaturaController {
  constructor(
    @Inject('IFaturaService')
    private readonly faturaService: IFaturaService,
  ) {}
  @Get('fatura-atual')
  async buscarItensFaturaAtual(
    @Query() paginationDTO: AppPaginationDTO,
    @User() usuario: UsuarioAutenticadoDto,
  ): Promise<Page<FaturaItem>> {
    const itens = await this.faturaService.buscarItensFaturaAtual(
      paginationDTO,
      usuario,
    );
    return itens;
  }

  /*
   * @todo nao testaqdo ainda na web
   * */
  @Post('fechar')
  async fechar(): Promise<void> {
    await this.faturaService.fecharFatura();
  }

  @Get('buscar-itens-fatura')
  async buscarItensFatura(
    @Query() paginationDTO: BuscarLancamentosFaturaRequestDTO,
    @User() usuario: UsuarioAutenticadoDto,
  ): Promise<ConsultaItensFaturaResponseDTO> {
    const itens = await this.faturaService.buscarItensFatura(
      paginationDTO,
      usuario,
    );
    const totalFatura = await this.faturaService.somarTotalFatura(
      paginationDTO,
      usuario,
    );
    const response = new ConsultaItensFaturaResponseDTO(totalFatura, itens);
    return response;
  }
}
